messaging/src/main/java/org/axonframework/common/caching/Cache.java

```diff
@@ -59,6 +60,20 @@ public interface Cache {
      */
     boolean putIfAbsent(Object key, Object value);
 
+    /**
+     * Returns the value under the given {@code key} in the cache. If there is no value present, will invoke the given
+     * {@code valueSupplier}, put the value in the cache and return the produced value.
+     *
+     * @param key           The key under which the item was cached. If not present, this key is used to cache the
+     *                      outcome of the {@code valueSupplier}.
+     * @param valueSupplier A supplier that lazily supplies the value if there's no {@code key} present.
+     * @return The value that is in the cache after the operation. This can be the original value or the one supplied by
+     * the {@code valueSupplier}.
+     */
+    default <T> T computeIfAbsent(Object key, Supplier<T> valueSupplier) {
+        throw new UnsupportedOperationException("Cache#computeIfAbsent is currently unsupported by this version");
```

Comment:
Since this API is used by the framework and expected to be implemented by components, using a default method that throws an exception is worse than not making a default implementation at all. Currently, problems would only surface at runtime. 
Also see my other comment, I believe an implementation that simply does a 'return if not null, otherwise put and return' would be best here

Reply:
I'd wager the chance extremely slim users implement their own `Cache` version. 
Granted, you're right (as you tend to be from time to time :stuck_out_tongue: ). 
If they do, this is terrible. I'll provide a workable default implementation of the method.